hazelcast/src/main/java/com/hazelcast/client/util/StaticLB.java

```diff
@@ -38,7 +38,7 @@ public void init(Cluster cluster, ClientConfig config) {
     }
 
     @Override
-    public Member next() {
-        return member;
+    public Member next(boolean dataMember) {
+        return dataMember && member.isLiteMember() ? null : member;
```

Comment:
I am not sure I understand what you mean. This is a static load balancer that always returns the same member. If this member doesn't happen to be a data member, we return `null`.

Reply:
I think it would be better to throw in this case. By setting the `StaticLB` one asks that all traffic goes to a specific member, but here we return null which in turn will go to some data member.

Reply:
This is OK, since traffic may go to another member for many reasons:
1. If smart routing is disabled
1. If the member that is set is no longer reachable
1. Now we add the third case - if the added member is not a data member

Reply:
In other words, there is no such semantics, that forces us to use a specific member. Load balancer is a hint.